draft-ietf-httpbis-priority.md

```diff
@@ -0,0 +1,748 @@
+supplementary urgency.
+
+Values between 2 and 6 are used to represent this urgency, to provide
+flexibility to the endpoints for giving some responses more or less precedence
+than others that belong to the supplementary group. {{merging}} explains how
+these values might be used.
+
+Clients SHOULD NOT use values 2 and 6.  Servers MAY use these values to
+prioritize a response above or below other supplementary responses.
+
+Clients MAY use values 3 to indicate that a request is given relatively high
+priority, or 5 to indicate relatively low priority, within the supplementary
+urgency group.
+
+For example, an image certain to be visible at the top of the page, might be
+assigned a value of 3 instead of 4, as it will have a high visual impact for the
+user.  Conversely, an asynchronously loaded JavaScript file might be assigned an
+urgency value of 5, as it is less likely to have a visual impact.
+
+When none of the considerations above is applicable, the value of 3 SHOULD be
+used.
+
+### background
+
+The background urgency (value 7) is used for responses of which the delivery can
+be postponed without having an impact on using other responses.
+
+As an example, the download of a large file in a web browser would be assigned
+the background urgency so it would not impact further page loads on the same
+connection.
+
+## incremental
+
+The incremental(`i`) parameter takes an sh-boolean as the value that indicates if
+a response can be processed incrementally, i.e. provide some meaningful output
+as chunks of the response arrive.
+
+The default value of the incremental parameter is `0`.
+
+A server SHOULD distribute the bandwidth of a connection between incremental
+responses that share the same urgency.
+
+A server SHOULD transmit non-incremental responses one by one, preferably in the
+order the requests were generated.  Doing so maximizes the chance of the client
+making progress in using the composition of the HTTP responses at the earliest
+moment.
+
+The following example shows a request for a JPEG file with the urgency parameter
+set to `4` and the incremental parameter set to `1`.
+
+~~~ example
+:method = GET
+:scheme = https
+:authority = example.net
+:path = /image.jpg
+priority = u=4, i=?1
+~~~
+
+## Defining New Parameters
+
+When attempting to extend priorities, care must be taken to ensure any use of
+existing parameters are either unchanged or modified in a way that is backwards
+compatible for peers that are unaware of the extended meaning.
+
+# The Priority HTTP Header Field
+
+The Priority HTTP header field can appear in requests and responses. A client
+uses it to specify the priority of the response. A server uses it to inform
+the client that the priority was overwritten. An intermediary can use the
+Priority information from client requests and server responses to correct or
+amend the precedence to suit it (see {{merging}}).
+
+The Priority header field is an end-to-end signal of the request priority from
```

Comment:
iiuc: if it's `end-to-end`, I'd use "user-agent" and "origin server" instead of client and server.